SensorCommunication/Sensor.cpp

```diff
@@ -2,27 +2,65 @@
 // Created by pbrink on 22/03/18.
 //
 
+#include <stdexcept>
 #include "Sensor.h"
 
 Sensor::Sensor(string name,
                string address,
-               float samplingFrequency,
+               unsigned long int samplingFrequency,
                ReadWrite action,
                string inputFileName,
                string outputFileName) :
-                name(name), address(address), samplingFrequency(samplingFrequency), action(action)
+                name(name), address(address), action(action)
 {
     // ensure that input and output filenames are properly specified when necessary
     if (action == ReadWrite::READ || action == ReadWrite::READ_WRITE) {
         if (isFilenameAllowed(outputFileName)) {
             this->outputFileName = outputFileName;
         } else {
-            throw new std::exception
+            throw std::invalid_argument("Invalid filename provided");
         }
     }
+    if (action == ReadWrite::WRITE || action == ReadWrite::READ_WRITE) {
+        if (isFilenameAllowed(inputFileName)) {
+            this->inputFileName = inputFileName;
+        } else {
+            throw std::invalid_argument("Invalid filename provided");
+        }
+    }
+
+    // validate sampling frequency
+    if (samplingFrequency <= SAMPLING_FREQUENCY_LOWER_LIMIT || samplingFrequency > SAMPLING_FREQUENCY_UPPER_LIMIT) {
+        throw std::invalid_argument("Invalid sampling frequency provided");
+    } else {
+        this->samplingFrequency = samplingFrequency;
+    }
 }
 
 bool Sensor::isFilenameAllowed(string filename) {
     // TODO: verification of good filenames and paths
     return true;
 }
+
+bool Sensor::setSamplingFrequency() {
+    return false;
+}
+
+bool Sensor::start() {
```

Comment:
In my head if a function like `start()` or `stop()` would return `true`, it would mean that it successfully started or stopped the sensor. So if I call start() when it's already running it would return `false` to say: "this isn't doing anything because the sensor has already started". Anyway it's not a big deal functionally but I just wanted to bring it up anyway :eyes: